### PR TITLE
Build shared library for Apple Silicon

### PR DESCRIPTION
**Pull request by rob-odwyer**
This adds support for an `aarch64` build of the shared library, and updates the NativeLibraryLoader to load it on macOS when the `os.arch` system property is `aarch64`.

Although it's pretty straightforward to build for this architecture, running the tests is a lot more complex so I've disabled them when cross-compiling for now. It looks like [bytecodealliance/wasmtime](https://github.com/bytecodealliance/wasmtime) uses qemu to solve this problem for some architectures, but it's not as simple as just setting a target.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -170,7 +176,7 @@ task copyJniLib(type: Copy) {
     def targetDir = rustTargetTriple() ?: ""
     from "wasmtime-jni/target/$targetDir/release"
     include '*.so', '*.dylib', "*.dll"
-    rename "^(lib)?wasmtime_jni", "\$1wasmtime_jni_${project.version}_${jniLibOsClassifier()}"
+    rename "^(lib)?wasmtime_jni", "\$1wasmtime_jni_${project.version}_${jniLibOsClassifier()}${jniLibArchClassifier()}"
```
Comment (rob-odwyer) [inline build.gradle:179]: This results in the following file path for the aarch64 shared lib:
```
jni-libs/libwasmtime_jni_0.15.0_macos_aarch64.dylib
```

File: build.gradle [hunk 2]
```diff
@@ -71,6 +72,11 @@ static def jniLibOsClassifier() {
     throw new RuntimeException("platform not supported: " + System.getProperty("os.name"))
 }
 
+static def jniLibArchClassifier() {
+    def target = rustTargetTriple() ?: ""
+    return target ? "_" + target.split("-")[0] : ""
```
Comment (kawamuray) [inline build.gradle:77]: This project will support multiple target architectures for one operating system after this PR. I think for consistency, we should change all shared objects to have consistent naming convention that includes architecture as well, not just for aarch64 + macos.

File: src/main/java/io/github/kawamuray/wasmtime/NativeLibraryLoader.java [hunk 3]
```diff
@@ -79,10 +79,10 @@ private static String libVersion() throws IOException {
 
     @AllArgsConstructor
     private enum Platform {
-        LINUX("linux","lib" , ".so"),
-        MACOS("macos","lib", ".dylib"),
-        WINDOWS("windows","",".dll")
-        ;
+        LINUX("linux", "lib", ".so"),
+        MACOS("macos", "lib", ".dylib"),
+        MACOS_AARCH64("macos_aarch64", "lib", ".dylib"),
```
Comment (kawamuray) [inline src/main/java/io/github/kawamuray/wasmtime/NativeLibraryLoader.java:84]: Maybe we better have `enum Os` and `enum Arch`, and make this `Platform` to hold these, for better organization?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -31,18 +41,27 @@ jobs:
...
         docker run --rm --volume $PWD:/build --workdir /build build-image ./gradlew build copyJniLib -x javadoc
+    # Otherwise, run gradle normally to build the JNI lib
+    - name: Build with Gradle
+      if: ${{ !startsWith(matrix.os, 'ubuntu') }}
+      run: ./gradlew build copyJniLib ${GRADLE_ARGS}
+    - name: List shared library files
+      run:
+        ls build/jni-libs
     - name: Save JNI lib output
-      if: startsWith(github.ref, 'refs/tags/')
```
Comment (kawamuray) [inline .github/workflows/ci.yml:45]: Is this change intended? This will attempt to upload the built jnilibs for non-tagged commits.

File: build.gradle (hunk 2 above)
Comment (rob-odwyer) [reply build.gradle:77]: Makes sense - I was going for minimal changes, but it's probably best to be consistent.

**Comment by rob-odwyer** [comment]
Updated the build.gradle and the loading logic to use the following format (grabbed these from a tagged build on the fork to confirm):
```
$ file jni-libs/*
jni-libs/libwasmtime_jni_0.17.0_linux_x86_64.so:     ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, BuildID[sha1]=fe48cb0c3ace952c7619e0e498399475cf0ddd98, with debug_info, not stripped
jni-libs/libwasmtime_jni_0.17.0_macos_aarch64.dylib: Mach-O 64-bit dynamically linked shared library arm64
jni-libs/libwasmtime_jni_0.17.0_macos_x86_64.dylib:  Mach-O 64-bit dynamically linked shared library x86_64
jni-libs/wasmtime_jni_0.17.0_windows_x86_64.dll:     PE32+ executable (DLL) (GUI) x86-64, for MS Windows
```

On #36 - I think my timing was just lucky! I didn't run into any cross-compilation issues, so it must have been fixed upstream.

**Review by kawamuray — APPROVED** [review]
LGTM, thanks!
